features/kafka/producer/src/main/java/org/opennms/features/kafka/producer/ProtobufMapper.java

```diff
@@ -220,16 +223,26 @@ public ProtobufMapper(EventConfDao eventConfDao, HwEntityDao hwEntityDao, Sessio
                 LOG.warn("An error occurred when building node criteria for node with id: {}." +
                         " The node foreign source and foreign id (if set) will be missing from the event with id: {}.",
                         event.getNodeid(), event.getDbid(), e);
+                // We only include the node id in the node criteria in when forwarding events
+                // since the event does not currently contain the fs:fid or a reference to the node object.
                 builder.setNodeCriteria(OpennmsModelProtos.NodeCriteria.newBuilder()
                         .setId(event.getNodeid()));
             }
-            // We only include the node id in the node criteria in when forwarding events
-            // since the event does not currently contain the fs:fid or a reference to the node object.
-            builder.setNodeCriteria(OpennmsModelProtos.NodeCriteria.newBuilder()
-                    .setId(event.getNodeid()));
+
         }
-        if (event.getInterface() != null) {
-            builder.setIpAddress(event.getInterface());
+
+        getString(event.getInterface()).ifPresent(builder::setIpAddress);
+
+        for (Parm parm : event.getParmCollection()) {
+            if (parm.getParmName() == null || parm.getValue() == null) {
+                continue;
+            }
+            String value = EventDatabaseConstants.escape(parm.getValue().getContent() == null ? "" :
```

Comment:
We don't need to escape here do we?

```suggestion
            String value = parm.getValue().getContent() == null ? "" : parm.getValue().getContent();
```